diff2html/diff2html.d.ts

```diff
@@ -0,0 +1,66 @@
+// Type definitions for diff2html
+// Project: https://github.com/rtfpessoa/diff2html
+// Definitions by: rtfpessoa <https://github.com/rtfpessoa/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace Diff2Html {
+    
+    export interface Options {
+        inputFormat?: string;
+        outputFormat?: string;
+        showFiles?: boolean;
+        matching?: string;
+        synchronisedScroll?: boolean;
+        matchWordsThreshold?: number;
+        matchingMaxComparisons?: number;
+    }
+
+    export interface Line {
+        content: string;
+        type: string;
+        oldNumber: number;
+        newNumber: number;
+    }
+
+    export interface Block {
+        oldStartLine: number;
+        oldStartLine2?: number;
+        newStartLine: number;
+        header: string;
+        lines: Line[];
+    }
+
+    export interface Result {
+        addedLines: number;
+        deletedLines: number;
+        isCombined: boolean;
+        isGitDiff: boolean;
+        oldName: string;
+        newName: string;
+        language: string;
+        blocks: Block[];
+        oldMode?: string;
+        newMode?: string;
+        deletedFileMode?: string;
+        newFileMode?: string;
+        isDeleted?: boolean;
+        isNew?: boolean;
+        isCopy?: boolean;
+        isRename?: boolean;
+        unchangedPercentage?: number;
+        changedPercentage?: number;
+        checksumBefore?: string;
+        checksumAfter?: string;
+        mode?: string;
+    }
+    
+    export interface Diff2Html {
+        getJsonFromDiff(input: string, configuration?: Options): Result;
```

Comment:
@DanielRosenwasser I do not think I understand you.

I know what you want. But it does not seem that what you are suggesting me to do will compile to valid JS code that will work with the module.

currently the code compiles to:
```js
var Diff2Html = require("diff2html")
var d2h = Diff2Html.Diff2Html
```

and in d2h will the the public api of my library.

If in the typescript definition I remove that when the code compiles it will not run since if I do just:
```js
var d2h = require("diff2html")
```

d2h will not have the methods. Do you understand my question? Or am I being somehow confusing?